bridge/svix-bridge/src/main.rs

```diff
@@ -2,18 +2,36 @@ use self::config::Config;
 use clap::Parser;
 use lazy_static::lazy_static;
 use opentelemetry::runtime::Tokio;
+use opentelemetry::sdk::export::metrics::aggregation::delta_temporality_selector;
+use opentelemetry::sdk::metrics::controllers::BasicController;
+use opentelemetry::sdk::metrics::selectors;
 use opentelemetry_otlp::WithExportConfig;
 use std::io::{Error, ErrorKind, Result};
 use std::path::PathBuf;
 use std::time::Duration;
 use svix_bridge_types::{SenderInput, TransformerJob};
 use svix_ksuid::{KsuidLike as _, KsuidMs};
+#[cfg(all(not(target_env = "msvc"), feature = "jemalloc"))]
+use tikv_jemallocator::Jemalloc;
+use tracing::Instrument;
 use tracing_subscriber::prelude::*;
 
+mod allocator;
 mod config;
+mod metrics;
 mod runtime;
 mod webhook_receiver;
 
+use crate::allocator::{get_allocator_stat_mibs, get_allocator_stats};
+use crate::metrics::CommonMetrics;
+
+#[cfg(all(not(target_env = "msvc"), feature = "jemalloc"))]
+#[global_allocator]
+static GLOBAL: Jemalloc = Jemalloc;
+
+#[cfg(all(target_env = "msvc", feature = "jemalloc"))]
+compile_error!("jemalloc cannot be enabled on msvc");
```

Comment:
I don't remember if the line ordering matters (I think it does?) but if we ensure this `compiler_error!()` invocation happens as early as possible, all the rest of the `cfg` attributes can simplify down to _just the feature_, I think.

Does that make sense?
Maybe this sort of trick only makes sense from a `build.rs` -- something that happens before the main sources are processed. 🤔

Reply:
I flipped the target_env to "gnu" and simplified the other conditions to try this out. Unfortunately it results in a variety of unresolved import errors.

Putting it in a build script works, but is still kind of a weird IDE experience. Compilation does exit with one nice clean error, but the unresolved import errors remain and _look_ like they're being used in the code.

The more complicated conditionals appear to be the best solution at the moment.

Reply:
Boo.